### PR TITLE
[bitnami/spark] Adds support for Spark metrics

### PR DESCRIPTION
**Pull request by rafariossaa**
**Description of the change**

Expose metrics for prometheus and prometheus-operator

**Benefits**

Get spark metrics in prometheus, issue #4429 

**Possible drawbacks**

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/spark/templates/statefulset-master.yaml [hunk 1]
```diff
@@ -17,8 +17,14 @@ spec:
         {{- if .Values.master.extraPodLabels }}
           {{- include "common.tplvalues.render" (dict "value" .Values.master.extraPodLabels "context" $) | nindent 8 }}
         {{- end }}
-      {{- if .Values.master.podAnnotations }}
-      annotations: {{- include "common.tplvalues.render" (dict "value" .Values.master.podAnnotations "context" $) | nindent 8 }}
+      {{- if or .Values.worker.podAnnotations .Values.metrics.enabled }}
+      annotations:
+      {{- if .Values.worker.podAnnotations }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.worker.podAnnotations "context" $) | nindent 8 }}
```
Comment (javsalgar) [inline bitnami/spark/templates/statefulset-master.yaml:23]: wouldn't it be `master.podAnnotations`?

File: bitnami/spark/templates/svc-worker-metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{{- if and .Values.metrics.enabled .Values.metrics.serviceMonitor.enabled }}
```
Comment (javsalgar) [inline bitnami/spark/templates/svc-worker-metrics.yaml:1]: I thought that the original service would work without the need of a new service, right?

**Review by javsalgar — CHANGES_REQUESTED** [review]

File: bitnami/spark/templates/statefulset-master.yaml (hunk 1 above)
Comment (rafariossaa) [reply bitnami/spark/templates/statefulset-master.yaml]: yes, you are right.

File: bitnami/spark/templates/svc-worker-metrics.yaml (hunk 2 above)
Comment (rafariossaa) [reply bitnami/spark/templates/svc-worker-metrics.yaml]: I move to PodMonitor, this way there is no necessity to create new services.

File: bitnami/spark/values-production.yaml [hunk 3]
```diff
@@ -259,6 +259,69 @@ worker:
...
+    ##
+    annotations:
+      prometheus.io/scrape: "true"
+      prometheus.io/port: "{{ .Values.metrics.service.port }}"
+
+  ## Prometheus Service Monitor
+  ## ref: https://github.com/coreos/prometheus-operator
+  ##      https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#endpoint
+  ##
+  serviceMonitor:
```
Comment (dani8art) [inline bitnami/spark/values-production.yaml:283]: We are saying `serviceMonitor` but we are using `PodMonitor` instead, could we align this?

Comment (rafariossaa) [reply bitnami/spark/values-production.yaml:283]: Sure, :facepalm:

**Review by dani8art — APPROVED** [review]

**Review by javsalgar — APPROVED** [review]

**Pull request merged**